connectors/sources/sharepoint_online.py

```diff
@@ -2457,4 +2481,18 @@ def _access_control_for_user(self, user):
         if email:
             user_access_control.append(_prefix_email(email))
 
+        if user_id:
+            user_access_control.append(_prefix_user_id(user_id))
+
         return user_access_control
+
+    async def _access_control_for_group_id(self, group_id):
+        if group_id.endswith("_o"):
```

Comment:
`group_id.endswith("_o")` it looks like a candidate for a separate function with a meaningful name and a comment explaining the logic behind the check